website/source/docs/jobspec/servicediscovery.html.md

```diff
@@ -0,0 +1,128 @@
+---
+layout: "docs"
+page_title: "Service Discovery in Nomad"
+sidebar_current: "docs-service-discovery"
+description: |-
+  Learn how to add service discovery to jobs
+  ---
+
+# Service Discovery
+
+Nomad enables dynamic scheduling on compute resources to run services at scale
+on compute nodes. Size of an application cluster varies depending
+on volume of traffic, health of services, etc thereby the network topology of
+services are constanly changing. This introduces the challenge of discovery of services, 
+Nomad solves this problem by integrating with [Consul](https://consul.io) to provide service
+discovery and health checks of services.
+
+Operators have to run Consul agents on a Nomad compute node. Nomad also makes
+running an application like Consul agent on every single node in a data centre
+simple by providing system jobs. Nomad connects to Consul on it's default http
+port but operators can override it.
+
+## Configuration
+
+* `consul.address`: This is a Nomad client config which can be used to override
+  the default Consul Agent HTTP port that Nomad uses to connect to Consul. The
+  default for this is "127.0.0.1:8500"
+
+## Service Deginition Syntax
+
+The service blocks in a Task definition defines a service which Nomad will
+register with Consul. Multiple Service blocks are allowed in a Task definition,
+which makes it usable for cases when an application listens on multiple ports
+each exposing a specific service.
+
+### Example 
+
+A brief example of a service definition in a Task
+```
+group "database" {
+    task "mysql" {
+        driver = "docker"
+        service {
+            tags = ["master", "mysql"]
+            port = "db"
+            check {
+                type = "tcp"
+                delay = "10s"
+                timeout = "2s"
+            }
+        }
+        reources {
+            cpu = 500
+            memory = 1024
+            network {
+                mbits = 10
+                port "db" {
+                }
+            }
+        }
+    }
+}
+
+```
+
+* `name`: Nomad automatically determines the name of a Task. By default the name
+  of a service is $(job-name)-$(task-group)-$(task-name). Users can explictly
```

Comment:
Might be worth to put it in the code syntax `$(job-name)-$(task-group)-$(task-name).`